modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/JavaInputResolverModelParser.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.runtime.module.extension.internal.loader.parser.java;
+
+import static org.mule.runtime.api.i18n.I18nMessageFactory.createStaticMessage;
+import static org.mule.runtime.core.api.util.ClassUtils.getClassName;
+
+import org.mule.runtime.api.exception.MuleRuntimeException;
+import org.mule.runtime.core.api.util.ClassUtils;
+import org.mule.runtime.module.extension.internal.metadata.SdkInputTypeResolverAdapter;
+import org.mule.sdk.api.metadata.resolving.InputTypeResolver;
+
+public class JavaInputResolverModelParser {
+
+  private final String parameterName;
+  private final Class<?> inputTypeResolverDeclarationClass;
+  private final boolean muleResolver;
+
+  public JavaInputResolverModelParser(String parameterName, Class<?> inputTypeResolverDeclarationClass, boolean muleResolver) {
+    this.parameterName = parameterName;
+    this.inputTypeResolverDeclarationClass = inputTypeResolverDeclarationClass;
+    this.muleResolver = muleResolver;
+  }
+
+  public String getParameterName() {
+    return parameterName;
```

Comment:
If the parser is attached to a parameter model parser, then this would not be necessary. Maybe when doing the decouple task we can delete this method.

Reply:
Yep